src/commons/eventsourcing/listener.rs

```diff
@@ -2,6 +2,16 @@ use std::sync::RwLock;
 
 use super::Aggregate;
 
+//------------ SyncEventListener ---------------------------------------------
```

Comment:
What is the meaning of the term "Sync" in this name? Given that both here and elsewhere you have to comment to say that a "Sync" event listener is invoked _before_ saving and a non-Sync listener is invoked _after_ saving, perhaps it would be better to name them differently, e.g. `PreSaveListener` and `PostSaveListener`?